packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -225,6 +229,25 @@ class SliverConstraints extends Constraints {
   /// contents depends on the [growthDirection].
   final double scrollOffset;
 
+  /// The scroll distance that has been consumed by all [Sliver]s that came
+  /// before this [Sliver].
+  ///
+  /// [Sliver]s are often created lazily as layout occurs, e.g., [SliverList].
+  /// In this case, when [Sliver]s exceed the viewport and are built lazily,
+  /// [consumedScrollExtent] will be [double.infinity] for all [Sliver]s that
```

Comment:
not necessarily. SliverList for example tries pretty hard to come up with an estimate even when it's not sure.

Reply:
Do you know what the requirements are to get that estimation?  I wrote this up after testing with the SliverList that I'm using for CupertinoTableView and it was reporting infinity until I got close to the end of the list (I assume it was infinity until I got with cacheExtent).

Reply:
Probably depends on what the delegate reports... for example if you give the count of children ahead of time